deep_translator/tencent.py

```diff
@@ -0,0 +1,139 @@
+"""
+tencent translator API
+"""
+
+__copyright__ = "Copyright (C) 2020 Nidhal Baccouri"
+
+import base64
+import hashlib
+import hmac
+import os
+import time
+from typing import List, Optional
+
+import requests
+
+from deep_translator.base import BaseTranslator
+from deep_translator.constants import (
+    BASE_URLS,
+    TENCENT_LANGUAGE_TO_CODE,
+    TENCENT_SECRET_ID_ENV_VAR,
+    TENCENT_SECRET_KEY_ENV_VAR,
+)
+from deep_translator.exceptions import (
+    ApiKeyException,
+    ServerException,
+    TencentAPIerror,
+    TranslationNotFound,
+)
+from deep_translator.validate import is_empty, is_input_valid
+
+
+class TencentTranslator(BaseTranslator):
+    """
+    class that wraps functions, which use the TentCentTranslator translator
+    under the hood to translate word(s)
+    """
+
+    def __init__(
+        self,
+        source: str = "en",
+        target: str = "zh",
+        secret_id: Optional[str] = os.getenv(TENCENT_SECRET_ID_ENV_VAR, None),
+        secret_key: Optional[str] = os.getenv(
+            TENCENT_SECRET_KEY_ENV_VAR, None
+        ),
+        **kwargs
+    ):
+        """
+        @param secret_id: your tencent cloud api secret id.
+        Get one here: https://console.cloud.tencent.com/capi
+        @param secret_key: your tencent cloud api secret key.
+        @param source: source language
+        @param target: target language
+        """
+        if not secret_id:
+            raise ApiKeyException(env_var=TENCENT_SECRET_ID_ENV_VAR)
+
+        if not secret_key:
+            raise ApiKeyException(env_var=TENCENT_SECRET_KEY_ENV_VAR)
+
+        self.secret_id = secret_id
+        self.secret_key = secret_key
+        url = BASE_URLS.get("TENENT")
+        super().__init__(
+            base_url=url,
+            source=source,
+            target=target,
+            languages=TENCENT_LANGUAGE_TO_CODE,
+            **kwargs
+        )
+
+    def translate(self, text: str, **kwargs) -> str:
+        """
+        @param text: text to translate
+        @return: translated text
+        """
+        if is_input_valid(text):
+            if self._same_source_target() or is_empty(text):
+                return text
+
+            # Create the request parameters.
+            translate_endpoint = "tmt.tencentcloudapi.com"
```

Comment:
Why not re-use this from BaseURL in constants.py? or create a variable for it in constants.py. IT would be easier to change it if something changes in the future